From 39403054d365dc7bcf0041e2f45a913bf83d53b7 Mon Sep 17 00:00:00 2001
From: Liu Haitao <haitao.liu@windriver.com>
Date: Mon, 18 Jun 2018 23:12:28 -0700
Subject: [PATCH] mtd:spi-nor:fix compile warning

fix commit 9cfa42b5fb93cae87(mtd: spi-nor: add stacked and dual-parallel support)

After applying this commit ,the usage of the function spi_nor_set_4byte_opcodes
has been changed as below:

	#if defined(CONFIG_ARCH_ZYNQMP) || defined(CONFIG_ARCH_ZYNQ)
		...
	#else
		...
		spi_nor_set_4byte_opcodes(nor, info);
		...
	#endif

when compile the kernel with CONFIG_ARCH_ZYNQMP and CONFIG_ARCH_ZYNQ, the
following wraning would be reported.

drivers/mtd/spi-nor/spi-nor.c:287:13: warning: 'spi_nor_set_4byte_opcodes'
defined but not used [-Wunused-function]

The spi_nor_set_4byte_opcodes does not be used in zynq and zynmp architectures.
So disable to compile it.

Signed-off-by: Liu Haitao <haitao.liu@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index e19814e..310d682 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -284,6 +284,10 @@ static inline u8 spi_nor_convert_3to4_erase(u8 opcode)
 				      ARRAY_SIZE(spi_nor_3to4_erase));
 }
 
+/*
+ *The ZYNQ AND ZYNQMP do not use spi_nor_set_4byte_opcodes any more.
+ * */
+#if !defined(CONFIG_ARCH_ZYNQMP) && !defined(CONFIG_ARCH_ZYNQ)
 static void spi_nor_set_4byte_opcodes(struct spi_nor *nor,
 				      const struct flash_info *info)
 {
@@ -303,6 +307,7 @@ static void spi_nor_set_4byte_opcodes(struct spi_nor *nor,
 	nor->program_opcode = spi_nor_convert_3to4_program(nor->program_opcode);
 	nor->erase_opcode = spi_nor_convert_3to4_erase(nor->erase_opcode);
 }
+#endif
 
 /* Enable/disable 4-byte addressing mode. */
 static inline int set_4byte(struct spi_nor *nor, const struct flash_info *info,
-- 
1.7.9.5

